### PR TITLE
Create GitHub Action to Run Cypress

### PR DESCRIPTION
**Pull request by BCerki**
This PR addresses issue:#101

Includes tests? y
Rebased on main address to conflicts? y
Updated docs? y

Proposed changes:
- Add a github action that runs the cypress test

Additional notes:
- There are some comments on an older version of this PR here: https://github.com/button-inc/digital_marketplace/pull/102
- Pre-testing, the action builds the whole app. The build action also builds the whole app--any way to avoid this duplication? Added this question to #104 to address later
- The action sets off Cypress errors but the test still runs so I think that's not a problem? They [mention this behaviour in their docs](https://docs.cypress.io/guides/references/troubleshooting#:~:text=Note%3A%20verbose%20Electron%20logging%20might%20show%20warnings%20that%20still%20allow%20Cypress%20to%20work%20normally.%20For%20example%2C%20the%20Cypress%20Test%20Runner%20opens%20normally%20despite%20the%20scary%20output%20below%3A): 
image

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: e2e
+on: [push]
+jobs:
+  cypress:
+    runs-on: ubuntu-latest
+    env:
+      POSTGRES_URL: "postgresql://digitalmarketplace:digitalmarketplace@localhost:5432/digitalmarketplace"
+      COOKIE_SECRET: "foobar"
+      MAILER_HOST: "foobar"
+      MAILER_PORT: "321"
```
Comment (BCerki) [inline .github/workflows/e2e.yml:10]: The app wouldn't start unless I had these filled out, but these placeholders seem to work

Comment (wenzowski) [reply .github/workflows/e2e.yml:10]: Tech debt: let's run something like [Sendria](https://github.com/msztolcman/sendria) so these configured values are real within the testing environment.

File: package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "cypress:test": "npm run back-end:build && npm run back-end:start & wait-on --timeout 60000 http://localhost:3000 & ./node_modules/.bin/cypress run",
+    "build": "npm run back-end:build & npm run front-end:build",
```
Comment (wenzowski) [inline package.json]: We don't need to daemonize the backend build, right?
```suggestion
    "build": "npm run back-end:build && npm run front-end:build",
```

**Review by wenzowski — CHANGES_REQUESTED** [review]
I don't think the daemonization was intended. Let's fix that, document tech debt as new issues, and ship this

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (BCerki) [reply .github/workflows/e2e.yml:10]: https://github.com/button-inc/digital_marketplace/issues/113

File: package.json (hunk 2 above)
Comment (BCerki) [reply package.json]: Nope, just a typo

**Review by wenzowski — APPROVED** [review]
Regarding the Additional Notes:
Let's make sure we've got a tech debt card for our CI/CD process that notes our e2e tests should be booting the candidate docker image that we intend to release to production (as a [12 factor app](https://12factor.net/), we should be able to boot the prod image with `development` or `test` environment flag set). The cypress errors appear to be due to the fact that we're running headless from inside a runner that does not have access to a gpu. There's probably a way to squelch these, so that should be tracked as tech debt as well. The `npm run browserslist --update-db` call [should probably also be part of our ci config](https://github.com/browserslist/browserslist/issues/492) so I'd track that as separate tech debt.

**Pull request merged**